sdk/search/Azure.Search.Documents/src/Generated/DataSourcesClient.cs

```diff
@@ -41,7 +41,17 @@ internal DataSourcesClient(ClientDiagnostics clientDiagnostics, HttpPipeline pip
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         public virtual async Task<Response<SearchIndexerDataSource>> CreateOrUpdateAsync(string dataSourceName, SearchIndexerDataSource dataSource, Guid? xMsClientRequestId = null, string ifMatch = null, string ifNoneMatch = null, CancellationToken cancellationToken = default)
         {
-            return await RestClient.CreateOrUpdateAsync(dataSourceName, dataSource, xMsClientRequestId, ifMatch, ifNoneMatch, cancellationToken).ConfigureAwait(false);
+            using var scope = _clientDiagnostics.CreateScope("DataSourcesClient.CreateOrUpdate");
```

Comment:
In Search, we call into the REST clients directly. This change breaks tracing. We don't need and shouldn't be forced to take the overhead of this wrapper class that we internalize anyway. Could you please move the scope creation back to the REST client?

Reply:
Search build seems to be passing.

What's the overhead?

Idea is that scopes are a an outer client concern because it wraps REST calls into logical operations, knows about the correct names (`StartOperation`, `GetFirstPage` vs `GetNextPage` etc.) and so forth.

Reply:
Not "breaking" as in we won't compile, but that events won't be traced for Search now. Right now, our public clients use `[ForwardedsClientCalls]` and call the `\w*RestClient` methods directly. If tracing is moved up to the `\w*(?<!Rest)Client` methods, no tracing will be performed at all.

Reply:
Okay, the fix would be to wrap non-rest clients. Do you want us to mange the change  as part of this PR?

Reply:
Why the overhead of the higher-level operations classes? They provide no benefit. Was there a reason the tracing was moved up a level? Seems putting it as low-level as possible would be safer.

Reply:
What overhead are you talking about?

Because diagnostic scopes are supposed to represent logical operations not rest calls and the non-rest clients wrap REST calls into the logical operations. 

Also diagnostic scope names are supposed to be the same as main client method names not rest client so when we add customizations those would be able to easily stay in sync.

Reply:
Then what is the point of `[ForwardsClientCalls]`? The generated operations clients were thin wrappers over the REST clients anyway. But we wanted more customization over the clients themselves and ended up internalizing them all.

/cc @tg-msft

Reply:
> Then what is the point of [ForwardsClientCalls]? 

It's a temporary workaround to suppress diagnostic name testing that would eventually have to be undone when we figure out entire diagnostic story.

Reply:
Rather than having to change all callers to create these scopes ourselves, could we instead just pass in the scope name as an optional parameter? If not passed, format the name as you are today. This would be a much easier change, and could still work with the higher-level generated clients, i.e. the *Client could pass in the method name (though, aren't they the same anyway?) to the *RestClient, and any manually created *Client classes could do the same. If the *RestClient creates the scope with the passed-in or formatted name, everyone wins.

Reply:
The scope name is not the only thing, the scope has to capture all operations that are executing in the method not only the rest call. 

For example if there is some custom mapping code in the client method and it throws an exception the scope should capture it. It's not possible is the scope is in RestClient.

Reply:
Since you're working on some refactoring to make this easier, then, can we hold off? This will leave Search with *no* distributed tracing.

Reply:
Do you plan to ship any time soon?

Reply:
BTW, we are willing to make fixes to add diagnostic scopes into Search, even if manually for now.

Reply:
It took a while, but I finally updated the Search methods to all have diagnostic scopes in them. Pavel's snippet helped out a lot, but I had issues getting it going because of my Visual Studio version. I had to uninstall it completely and reinstall the release version. Anyway, it is done. Let me know if there are issues.